android/openthread_commissioner/service/src/main/java/io/openthread/commissioner/service/ThreadCommissionerServiceImpl.java

```diff
@@ -0,0 +1,364 @@
+      throw new ThreadCommissionerException(error.getCode(), error.getMessage());
+    }
+  }
+
+  @Override
+  public String onJoinerRequest(ByteArray joinerId) {
+    Log.d(
+        TAG,
+        String.format(
+            "A joiner (ID=%s) is requesting commissioning",
+            CommissionerUtils.getHexString(joinerId)));
+
+    if (intermediateStateCallback != null) {
+      intermediateStateCallback.onJoinerRequest(CommissionerUtils.getByteArray(joinerId));
+    }
+
+    if (matchJoinerId(getCurJoinerId(), joinerId)) {
+      return curJoinerInfo.getPskd();
+    }
+    return "";
+  }
+
+  @Override
+  public void onJoinerConnected(ByteArray joinerId, Error error) {
+    Log.d(TAG, "A joiner is connected");
+  }
+
+  @Override
+  public boolean onJoinerFinalize(
+      ByteArray joinerId,
+      String vendorName,
+      String vendorModel,
+      String vendorSwVersion,
+      ByteArray vendorStackVersion,
+      String provisioningUrl,
+      ByteArray vendorData) {
+    Log.d(
+        TAG,
+        String.format("A joiner (ID=%s) is finalizing", CommissionerUtils.getHexString(joinerId)));
+
+    if (matchJoinerId(getCurJoinerId(), joinerId)) {
+      curJoinerCommissioned.open();
+    }
+
+    return true;
+  }
+
+  @Override
+  public void onKeepAliveResponse(Error error) {
+    Log.d(TAG, "received keep-alive response: " + error.toString());
+  }
+
+  @Override
+  public void onPanIdConflict(String peerAddr, ChannelMask channelMask, int panId) {
+    Log.d(TAG, "received PAN ID CONFLICT report");
+  }
+
+  @Override
+  public void onEnergyReport(String aPeerAddr, ChannelMask aChannelMask, ByteArray aEnergyList) {
+    Log.d(TAG, "received ENERGY SCAN report");
+  }
+
+  @Override
+  public void onDatasetChanged() {
+    Log.d(TAG, "Thread Network Dataset chanaged");
+  }
+
+  private ByteArray getCurJoinerId() {
+    if (curJoinerInfo == null) {
+      return null;
+    }
+
+    return Commissioner.computeJoinerId(new BigInteger(curJoinerInfo.getEui64()));
+  }
+
+  private boolean matchJoinerId(ByteArray curJoinerId, ByteArray inputJoinerId) {
+    if (curJoinerId == null
+        || inputJoinerId == null
+        || curJoinerId.size() != inputJoinerId.size()) {
+      return false;
+    }
+    for (int i = 0; i < curJoinerId.size(); ++i) {
+      if (curJoinerId.get(i) != inputJoinerId.get(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private static String getBorderAgentAddress(BorderAgentInfo borderAgentInfo) {
+    InetAddress addr = borderAgentInfo.host;
+
+    String ret = addr.getHostAddress();
+    if (addr instanceof Inet6Address) {
+      Inet6Address ipv6Addr = (Inet6Address) addr;
+      if (ipv6Addr.isLinkLocalAddress()) {
+        // FIXME(wgtdkp): This is just a workaround. The Border Agent address returned
+        //  by the NSDManager should already include scope ID (or interface name) for
+        //  link-local IPv6 Address.
+        ret += "%wlan0";
```

Comment:
any plan to fix it?

Reply:
No easy way in my mind for now. This workaround should work at least.